lightning/src/routing/router.rs

```diff
@@ -4956,6 +5109,104 @@ mod tests {
 		}
 	}
 
+	#[test]
+	fn adds_and_limits_cltv_offset() {
+		let (secp_ctx, network_graph, _, _, logger) = build_graph();
+		let (_, our_id, _, nodes) = get_nodes(&secp_ctx);
+
+		let scorer = test_utils::TestScorer::with_penalty(0);
+
+		let payment_params = PaymentParameters::from_node_id(nodes[6]).with_route_hints(last_hops(&nodes));
+		let keys_manager = test_utils::TestKeysInterface::new(&[0u8; 32], Network::Testnet);
+		let random_seed_bytes = keys_manager.get_secure_random_bytes();
+		let route = get_route(&our_id, &payment_params, &network_graph.read_only(), None, 100, 42, Arc::clone(&logger), &scorer, &random_seed_bytes).unwrap();
+		let cltv_expiry_deltas_before = route.paths.iter().flat_map(|p| p.iter().map(|h| h.cltv_expiry_delta).collect::<Vec<u32>>()).collect::<Vec<u32>>();
+
+		// Check whether the offset added to the last hop by default is in [1 .. DEFAULT_MAX_TOTAL_CLTV_EXPIRY_DELTA]
+		let mut route_default = route.clone();
+		add_random_cltv_offset(&mut route_default, &payment_params, &network_graph.read_only(), &random_seed_bytes);
+		let cltv_expiry_deltas_default = route_default.paths.iter().flat_map(|p| p.iter().map(|h| h.cltv_expiry_delta).collect::<Vec<u32>>()).collect::<Vec<u32>>();
+		assert_eq!(cltv_expiry_deltas_before[.. cltv_expiry_deltas_default.len()-1], cltv_expiry_deltas_default[.. cltv_expiry_deltas_default.len()-1]);
+		assert!(cltv_expiry_deltas_default.last() > cltv_expiry_deltas_before.last());
+		assert!(cltv_expiry_deltas_default.last().unwrap() <= &DEFAULT_MAX_TOTAL_CLTV_EXPIRY_DELTA);
```

Comment:
Could simplify this a big by using `split_last`.

Reply:
Yep, simplified this in https://github.com/lightningdevkit/rust-lightning/pull/1286/commits/0b777cbd4ac95d1578264dab82826d725d2cd3d3.